include/os/linux/spl/sys/kmem.h

```diff
@@ -43,10 +43,12 @@ extern void kmem_strfree(char *str);
 #define	KM_SLEEP	0x0000	/* can block for memory; success guaranteed */
 #define	KM_NOSLEEP	0x0001	/* cannot block for memory; may fail */
 #define	KM_PUSHPAGE	0x0004	/* can block for memory; may use reserve */
+#define	KM_ONCE		0x0008	/* like KM_SLEEP but do not loop allocation */
 #define	KM_ZERO		0x1000	/* zero the allocation */
 #define	KM_VMEM		0x2000	/* caller is vmem_* wrapper */
+#define	KM_KVMEM	0x4000	/* caller is kvmem_* wrapper */
 
-#define	KM_PUBLIC_MASK	(KM_SLEEP | KM_NOSLEEP | KM_PUSHPAGE)
+#define	KM_PUBLIC_MASK	(KM_SLEEP | KM_NOSLEEP | KM_PUSHPAGE | KM_ONCE)
```

Comment:
Are these changes still needed or can be cleaned up?

Reply:
can be cleanup up, even if i think its a usefull feature to prevent deadlocks in zfs

Reply:
Yeah, but that would be a (albeit usefull) seperate PR indeed.